src/pages/ContributorFields.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TextField, Typography } from '@material-ui/core';
+import { Contributor, Resource } from '../types/resource.types';
+import { ErrorMessage, Field, FieldArray, FieldArrayRenderProps, FieldProps, FormikProps, FormikValues } from 'formik';
+import Button from '@material-ui/core/Button';
+import { createContributor, putContributorFeature } from '../api/resourceApi';
+import Paper from '@material-ui/core/Paper';
+import styled from 'styled-components';
+import DeleteIcon from '@material-ui/icons/Delete';
+import AddIcon from '@material-ui/icons/Add';
+
+const StyledPaper = styled(Paper)`
+  width: 100%;
+  padding: 3rem;
+`;
+
+const StyledField = styled(Field)`
+  display: inline-block;
+  margin-left: 2rem;
+  margin-right: 2rem;
+  padding: 1rem;
+`;
+const StyledDiv = styled.div`
+  display: inline-block;
+  margin-top: 1rem;
+  margin-bottom: 0.3rem;
+`;
+
+const StyledTextField = styled(TextField)`
+  width: 22rem;
+  margin-left: 0.5rem;
+  margin-right: 0.5rem;
+`;
+
+const StyledButton = styled(Button)`
+  position: relative;
+  vertical-align: bottom;
+`;
+
+interface ContributorFieldsProps {
+  resource: Resource;
+  formikProps: FormikProps<FormikValues>;
+  setAllChangesSaved: any;
+}
+
+const ContributorFields: FC<ContributorFieldsProps> = ({ resource, formikProps, setAllChangesSaved }) => {
+  const { t } = useTranslation();
+
+  const addContributor = (arrayHelpers: FieldArrayRenderProps) => {
+    setAllChangesSaved(false);
+    createContributor(resource.identifier).then((contributorResponse) => {
+      arrayHelpers.push({
+        identifier: contributorResponse.data.identifier,
+        features: {
+          dlr_contributor_name: '',
+          dlr_contributor_type: '',
+          dlr_contributor_identifier: contributorResponse.data.identifier,
+        },
+      });
+      setAllChangesSaved(true);
+    });
+  };
+
+  const saveContributorField = async (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>,
+    resetForm: any,
+    currentValues: FormikValues,
+    index: number
+  ) => {
+    setAllChangesSaved(false);
+    const name = '' + event.target.name.split('.').pop();
+    if (event.target.value.length > 0) {
+      const contributorIdentifier = formikProps.values.resource?.contributors[index]?.identifier;
```

Comment:
ser at samme data er tilgjengelig flere plasser nå, dette er feks det samme:
```suggestion
      const contributorIdentifier = currentValues.resource?.contributors[index]?.identifier;
```
I NVA har vi droppet å sende `formikProps` etc rundt omkring som prop også, men heller brukt [useFormikContext()](https://formik.org/docs/api/useFormikContext) for å få tilgang til hva vi trenger der vi trenger det. Da slipper dere både `currentValues` som argument her og `formikProps` som prop til komponenten

Reply:
Fikset